azure/services/asogroups/spec.go

```diff
@@ -55,6 +55,9 @@ func (s *GroupSpec) Parameters(ctx context.Context, object genruntime.MetaObject
 	return &asoresourcesv1.ResourceGroup{
 		ObjectMeta: metav1.ObjectMeta{
 			OwnerReferences: []metav1.OwnerReference{s.Owner},
+			// Annotations:     map[string]string{
+			//   "serviceoperator.azure.com/credential-from": "my-resource-secret",
```

Comment:
Is there a way we can more generically annotate ASO resources with the secret in azure/services/aso/aso.go without needing to explicitly specify it here?

Reply:
It might be, I was thinking the same. So my idea is to grab the `spec.identityRef` of the cluster((i.e. AzureClusters/AzureManagedControlPlanes) in `aso.go`, compute the secret name and label the ASO object with the  "serviceoperator.azure.com/credential-from": "<secret_name>"

Reply:
Added the changes in https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3698/commits/34fdf794228d703d638522c2a258614e8f61e0f5